blob-polyfill.ts

```diff
@@ -0,0 +1,99 @@
+import { getArrayBufferForBlob } from "react-native-blob-jsi-helper";
+import { ReadableStream } from "web-streams-polyfill";
+
+const decoder = new TextDecoder();
+
+/**
+ * React Native's Blob implementation has a constructor that currently only supports
+ * constructing from parts that are of type: Array<Blob | String>.
+ *
+ * Web5 packages need to be able to construct a Blob where one of the parts is of type `Uint8Array`.
+ *
+ * This function updates the constructor of Blob to convert any parts that are of type `Uint8Array`
+ * into a `string`, so that it can properly work with React Native's Blob implementation.
+ */
+function monkeyPatchBlobConstructor() {
+  const OriginalBlob = global.Blob;
+
+  const blobProxyHandler = {
+    construct(target: any, argumentsList: any) {
+      const blobParts = argumentsList[0];
+      const options = argumentsList[1];
+
+      if (blobParts) {
+        for (const [index, element] of blobParts.entries()) {
+          if (element instanceof Uint8Array) {
+            blobParts[index] = decoder.decode(element);
+          }
+        }
+      }
+
+      return new target(blobParts, options);
+    },
+  };
+
+  const blobProxy = new Proxy(OriginalBlob, blobProxyHandler);
+  (global as any).Blob = blobProxy;
```

Comment:
I don't think you need to cast here. You can declare the global either as a declaration and/or in tsconfig